### PR TITLE
ISPN-5626 Clean up missing parts for Distributed Streams

### PR DESCRIPTION
**Pull request by wburns**
* Fixed various issues with cache loader
* Added a bunch of tests ported from Entry Retriever

https://issues.jboss.org/browse/ISPN-5630

### CODE REVIEW

File: core/src/test/java/org/infinispan/stream/stress/DistributedStreamRehashStressTest.java [hunk 1]
```diff
@@ -44,7 +44,7 @@
  * @author wburns
  * @since 8.0
  */
-@Test(groups = "stress", testName = "stream.stress.DistributedStreamRehashStressTest")
+@Test(groups = "stress", testName = "stream.stress.DistributedStreamRehashStressTest", enabled = false)
```
Comment (danberindei) [inline core/src/test/java/org/infinispan/stream/stress/DistributedStreamRehashStressTest.java]: Disabled tests should have a description.

Comment (wburns) [inline core/src/test/java/org/infinispan/stream/stress/DistributedStreamRehashStressTest.java]: Whoops it wasn't meant to be disabled.  I only did that so I could run the entire package more easily.  Thanks :)

**Comment by wburns** [comment]
Added back in test that was disabled on accident.

**Comment by wburns** [comment]
CI passed earlier http://ci.infinispan.org/viewLog.html?buildId=28489&tab=buildResultsDiv&buildTypeId=bt9

**Comment by gustavonalle** [comment]
testing

File: core/src/test/java/org/infinispan/stream/BaseClusteredStreamIteratorTest.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      Iterator<Map.Entry<Object, String>> iter = values.entrySet().iterator();
+      Map.Entry<Object, String> excludedEntry = iter.next();
+      // Remove it so comparison below will be correct
+      iter.remove();
+
+      Cache<MagicKey, String> cache = cache(1, CACHE_NAME);
+
+      Iterator<CacheEntry<MagicKey, String>> iterator = cache.getAdvancedCache().cacheEntrySet().stream().filter(
+              CacheFilters.predicate(new KeyFilterAsKeyValueFilter<>(new CollectionKeyFilter<>(
+                              Collections.singleton(excludedEntry.getKey()))))).iterator();
```
Comment (danberindei) [inline core/src/test/java/org/infinispan/stream/BaseClusteredStreamIteratorTest.java]: No lambdas??

File: core/src/test/java/org/infinispan/stream/BaseStreamIteratorEvictionTest.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      // First put some values in there
+      Map<Object, String> valuesInserted = new LinkedHashMap<Object, String>();
+      for (int i = 0; i < 5; ++i) {
+         Object key = i;
+         String value = key + " stay in cache";
+         cache.put(key, value);
+         valuesInserted.put(key, value);
+      }
+
+      int expectedTime = 2;
```
Comment (danberindei) [inline core/src/test/java/org/infinispan/stream/BaseStreamIteratorEvictionTest.java]: Lately I've been suggesting names like `LIFESPAN_SECONDS`.

File: core/src/test/java/org/infinispan/stream/BaseClusteredStreamIteratorTest.java (hunk 2 above)
Comment (wburns) [inline core/src/test/java/org/infinispan/stream/BaseClusteredStreamIteratorTest.java]: Not sure I know what you mean.

This was just copied from https://github.com/infinispan/infinispan/blob/master/core/src/test/java/org/infinispan/iteration/BaseClusteredEntryRetrieverTest.java#L71 and tweaked to use stream instead.

File: core/src/test/java/org/infinispan/stream/BaseStreamIteratorEvictionTest.java (hunk 3 above)
Comment (wburns) [inline core/src/test/java/org/infinispan/stream/BaseStreamIteratorEvictionTest.java]: Yeah actually this test should be tweaked to use TimeService instead.  I can add that as a TODO in the other issue.

**Comment by danberindei** [comment]
Looks good to me.

**Comment by gustavonalle** [comment]
pulling

**Comment by gustavonalle** [comment]
merged, tks @wburns !

**Pull request closed without merging**